test/junit/scala/util/SortingTest.scala

```diff
@@ -34,10 +36,17 @@ class SortingTest {
     val sxs = { val temp = xs.clone; Sorting.stableSort(temp); temp }
     val rxs = { val temp = xs.clone; Sorting.stableSort(temp)(backwardsN); temp }
     val sys = Sorting.stableSort(ys.clone.toIndexedSeq, (i: Int) => xs(i))
+
+    val xxs = Array.fill(size)(rng.nextInt(variety))
+    val sortSliceArray = xxs.clone(); Sorting.stableSort(sortSliceArray, xxs.length / 2, xxs.length)
+    val sliceMin = xxs.slice(xxs.length / 2, xxs.length).minOption.getOrElse(Int.MinValue)
+    val sliceAboveMinCountsEqual = xxs.slice(xxs.length / 2, xxs.length).count(_ >= sliceMin) == sortSliceArray.slice(xxs.length / 2, xxs.length).count(_ >= sliceMin)
```

Comment:
could we also check that the first half of the arrays are identical?

Reply:
@bal-a send a ping once that's pushed, then I'll merge the PR. You can squash into the second commit. Thanks!

Reply:
Thank you both for the feedback - added it and squashed it with the second commit, let me know if something looks off :)

@lrytz ping